src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/controllers/NotificationsController.java

```diff
@@ -119,6 +119,23 @@
         return ResponseEntity.ok(AboutToStartOrSubmitCallbackResponse.builder().data(caseData).build());
     }
 
+    @PostMapping(value = "/case-orchestration/notify/prepare-for-hearing", consumes = APPLICATION_JSON_VALUE)
+    @ApiOperation(value = "send e-mail for 'Prepare for hearing'.")
+    @ApiResponses(value = {
+        @ApiResponse(code = 204, message = "Prepare for hearing e-mail sent successfully",
+            response = AboutToStartOrSubmitCallbackResponse.class)})
+    public ResponseEntity<AboutToStartOrSubmitCallbackResponse> sendPrepareForHearingEmail(
+        @RequestBody CallbackRequest callbackRequest) {
+        log.info("Received request to send email to Solicitor for 'Prepare for hearing' for Case ID: {}", callbackRequest.getCaseDetails().getId());
+        validateCaseData(callbackRequest);
+        Map<String, Object> caseData = callbackRequest.getCaseDetails().getData();
+        if (isSolicitorAgreedToReceiveEmails(caseData, SOLICITOR_AGREE_TO_RECEIVE_EMAILS)) {
+            log.info("Sending email notification to Solicitor for 'Prepare for hearing'");
+            notificationService.sendPrepareForHearingEmail(callbackRequest);
+        }
+        return ResponseEntity.ok(AboutToStartOrSubmitCallbackResponse.builder().data(caseData).build());
+    }
+
     private boolean isSolicitorAgreedToReceiveEmails(Map<String, Object> mapOfCaseData, String solicitorAgreeToReceiveEmails) {
```

Comment:
I have a feeling this also may go to CommonFunctions and re-used.

Can you please check out if we have used it somewhere else?